src/include/uapi/ipc/topology.h

```diff
@@ -162,6 +163,21 @@ struct sof_ipc_comp_volume {
 	uint32_t initial_ramp;	/**< ramp space in ms */
 } __attribute__((packed));
 
+/* generic selector component */
+struct sof_ipc_comp_selector {
+	struct sof_ipc_comp comp;
+	struct sof_ipc_comp_config config;
+
+	/* selector supports 1 input and 1 output */
+	uint32_t input_channels_count;	/**< accepted values 2 or 4 */
+	uint32_t output_channels_count;	/**< accepted values 1 or 2 or 4 */
```

Comment:
The limitations for channels count should be removed here (since the selector could be updated to support more). Any request for > channels than supported should return an error at topology load time.
Selecting source and sink channels should be done using masks as I dont see how the current IPC structure would allow mapping 4 source -> 2 sink channels.